templates/kong.yml.erb

```diff
@@ -1,394 +1,102 @@
-## Used internally by this Kong node. Only IPv4 addresses are allowed (no hostnames).
-# cluster_listen_rpc: "127.0.0.1:7373"
-cluster_listen_rpc: '<%= @cluster_listen_rpc_address %>:<%= @cluster_listen_rpc_port %>'
+cluster_listen: "<%= @cluster_listen -%>"
 
-######
-## The path to the SSL certificate and key that Kong will use when listening on the `https` port.
-# ssl_cert_path: /path/to/certificate.pem
-# ssl_key_path: /path/to/certificate.key
-<% if not @ssl_cert_path.nil? and @ssl_cert_path -%>
-ssl_cert_path: <%= @ssl_cert_path %>
+cluster_listen_rpc: "<%= @cluster_listen_rpc -%>"
+
+<% if @ssl_cert_path -%>
+ssl_cert_path: "<%= @ssl_cert_path -%>"
 <% end -%>
 
-<% if not @ssl_key_path.nil? and @ssl_key_path -%>
-ssl_key_path: <%= @ssl_key_path %>
+<% if @ssl_key_path -%>
+ssl_key_path: "<%= @ssl_key_path -%>"
 <% end -%>
 
-######
-## Specify how Kong performs DNS resolution (in the `dns_resolvers_available` property) you want to use.
-## Options are: "dnsmasq" (You will need dnsmasq to be installed) or "server".
-# dns_resolver: dnsmasq
-dns_resolver: '<%= @dns_resolver %>'
+dns_resolver: "<%= @dns_resolver -%>"
 
-######
-## A dictionary of DNS resolvers Kong can use, and their respective properties.
-## Currently `dnsmasq` (default, http://www.thekelleys.org.uk/dnsmasq/doc.html) and `server` are supported.
-## By choosing `dnsmasq`, Kong will resolve hostnames using the local `/etc/hosts` file and `resolv.conf`
-## configuration. By choosing `server`, you can specify a custom DNS server.
-# dns_resolvers_available:
-  # server:
-    # address: "8.8.8.8:53"
-  # dnsmasq:
-    # port: 8053
-# FIXME Make this a parameter and loop through hash
 dns_resolvers_available:
-  dnsmasq:
-    port: 8053
-
-######
-## Cluster settings between Kong nodes.
-## For more information take a look at the Clustering Reference: https://getkong.org/docs/latest/clustering/
-# cluster:
-
-  ######
-  ## Address and port used by the node to communicate with other Kong nodes in the cluster with both UDP and
-  ## TCP messages. All the nodes in the cluster must be able to communicate with this node on this address.
-  ## Only IPv4 addresses are allowed (no hostnames).
-  ## The advertise flag is used to change the address that we advertise to other nodes in the
-  ## cluster. By default, the cluster_listen address is advertised. If the cluster_listen host is "0.0.0.0",
-  ## then the first local, non-loopback, IPv4 address will be advertised to the other nodes. However, in some
-  ## cases (specifically NAT traversal), there may be a routable address that cannot be bound to. This flag
-  ## enables gossiping a different address to support this.
-  # advertise: ""
-
-  ######
-  ## Key for encrypting network traffic within Kong. Must be a base64-encoded 16-byte key.
-  # encrypt: "foo"
+<% @dns_resolvers_available.each do |resolver, values| -%>
+  <%= resolver -%>:
+<% values.each do |k,v| -%>
+    <%= k %>: "<%= v %>"
```

Comment:
Same problem here. This fails for integers as well, as they are set as strings. In this case `dns_resolvers_available.dnsmasq.port`